doc/move_group_interface/src/move_group_interface_tutorial.cpp

```diff
@@ -32,169 +32,481 @@
+  {
+    // Remove old paths and text
+    visual_tools_.deleteAllMarkers();
+    // To publish a text message inside Rviz
+    visual_tools_.publishText(text_pose_, msg, rvt::WHITE, rvt::XLARGE);
+    // To publish a planed path
+    visual_tools_.publishTrajectoryLine(plan.trajectory_, joint_model_group_);
+    // Batch publishing is used to reduce the number of messages being sent to RViz for large visualizations
+    visual_tools_.trigger();
+    // This prompts the user before continuing.
+    visual_tools_.prompt("Press 'next' in the window to continue");
+  }
+
+  /**
+  * Publishes the given message above the robot, displays a path, and points
+  **/
+  void publishToRViz(std::string msg, const moveit::planning_interface::MoveGroupInterface::Plan& plan, std::vector<geometry_msgs::Pose> points)
+  {
+    // BEGIN_SUB_TUTORIAL visualization
+
+    // There are many different ways to publish information to RViz. These are some commonly used commands.
+    //
+    // Remove old paths and text.
+    visual_tools_.deleteAllMarkers();
+    // To publish a text message inside Rviz.
+    visual_tools_.publishText(text_pose_, msg, rvt::WHITE, rvt::XLARGE);
+    // To publish a planed path.
+    visual_tools_.publishTrajectoryLine(plan.trajectory_, joint_model_group_);
+    // To publish points.
+    for( geometry_msgs::Pose point : points )
+      visual_tools_.publishAxisLabeled(point, "pose");
+    // Batch publishing is used to reduce the number of messages being sent to RViz for large visualizations.
+    // This publishes all the changes just added.
+    visual_tools_.trigger();
+    // This prompts the user before continuing.
+    visual_tools_.prompt("Press 'next' in the window to continue");
+    // END_SUB_TUTORIAL
+  }
+
+  /**
+  * Publishes the given path without removing old content.
+  */
+  void publishPath(const moveit::planning_interface::MoveGroupInterface::Plan& plan)
+  {
+    // To publish the path
+    visual_tools_.publishTrajectoryLine(plan.trajectory_, joint_model_group_);
+    visual_tools_.trigger();
+    visual_tools_.prompt("Press 'next' in the window to continue");
+  }
+
+  /**
+  * Prints off basic information about move_group_
+  **/
+  void printBasicInfo()
+  {
+    // BEGIN_SUB_TUTORIAL basic_info
+    //
+    // Getting Basic Information
+    // ^^^^^^^^^^^^^^^^^^^^^^^^^
+    //
+    // move_group holds a lot of information about the robot.
+    //
+    // For example we can print the name of the reference frame for this robot.
+    ROS_INFO_NAMED(name_, "Planning frame: %s", move_group_.getPlanningFrame().c_str());
+
+    // We can also print the name of the end-effector link for this group.
+    ROS_INFO_NAMED(name_, "End effector link: %s", move_group_.getEndEffectorLink().c_str());
+
+    // We even can get a list of all the groups in the robot.
+    ROS_INFO_NAMED(name_, "Available Planning Groups:");
+    std::copy(move_group_.getJointModelGroupNames().begin(), move_group_.getJointModelGroupNames().end(),
```

Comment:
Don't print to cout if you can use the ros console functionality. 
Better to use string stream and pass that to `ROS_INFO_NAMED(name_, ss.toString())` <--- likely incorrect syntax